lxd/profiles_utils.go

```diff
@@ -232,3 +232,21 @@ func getProfileInstancesInfo(cluster *db.Cluster, projectName string, profileNam
 
 	return instances, nil
 }
+
+func validateProfileConfig(d *Daemon, config map[string]string) error {
```

Comment:
Why can't you do this as part of the existing `ValidConfig` function (just wrapped in an if for `instancetype.Any`)? Because with this change, you're now skipping a bunch of extra checks that occur in `ValidConfig` that are not in here? It would be good to keep instance validation (be it for profile or real instances) together.

Reply:
I thougth because i's profile specific it should be placed in `profiles_utils.go`, and its calling `ValidConfig` at `validateProfileConfig` also. Should i move the proflie validation logic back to  `ValidConfig` ?

Reply:
It would make sense, except that in doing that you're removed all this validation for profiles:

https://github.com/lxc/lxd/blob/6ad8779687c0987bfa7bbfe1ac3e5f547e1b7b93/lxd/instance/instance_utils.go#L149-L201

Reply:
sry, i don"t get it. `ValidateConfig` is called when. The part you highlighted should be executed or am I wrong?

https://github.com/lxc/lxd/blob/5d14dbfc368443c3154c36f7c0bf8fa86aca29b2/lxd/profiles_utils.go#L236-L240

Reply:
Oh I see you're calling instance.ValidConfig from inside `validateProfileConfig` I had thought it was instead of. OK that makes sense then to leave as is. As a general comment it would be easier to follow if you added some comments to your code to explain the logic being applied as I've found some of it non-trivial to reason about. Thanks

Reply:
ok shure, no problem :)